### PR TITLE
Feature/bootstrap update

### PR DESCRIPTION
**Pull request by iStrike7**

### CODE REVIEW

**Comment by iStrike7** [comment]
@mateuszaaa 
Please check that
1. Whitelist phase being equal to zero is fine
2. Default ratio is correct

File: pallets/bootstrap/src/lib.rs [hunk 1]
```diff
@@ -231,35 +234,35 @@ pub mod pallet {
...
+		// 		amount,
+		// 		ProvisionKind::Vested(
+		// 			vesting_starting_block.saturated_into::<BlockNrAsBalance>(),
+		// 			vesting_ending_block_as_balance.into(),
+		// 		),
+		// 	)?;
+		// 	ProvisionAccounts::<T>::insert(&sender, ());
+		// 	Self::deposit_event(Event::Provisioned(token_id, amount));
+		// 	Ok(().into())
+		// }
```
Comment (mateuszaaa) [inline pallets/bootstrap/src/lib.rs:265]:
```suggestion
```
remove comments

File: pallets/bootstrap/src/lib.rs [hunk 2]
```diff
@@ -826,12 +870,22 @@ impl<T: Config> Pallet<T> {
...
+					log!(
+						error,
+						"Activating liquidity tokens failed upon bootstrap claim rewards = ({:?}, {}, {})",
+						who,
+						liq_token_id,
+						non_vested_rewards
+					);
+					
+					Self::deposit_event(Event::RewardsLiquidityAcitvationFailed(who.clone(), liq_token_id, non_vested_rewards));
+				};
```
Comment (mateuszaaa) [inline pallets/bootstrap/src/lib.rs:888]: consider logging activate result error

```suggestion



				if let Err(err) = let activate_result = <T as Config>::RewardsApi::activate_liquidity_tokens(
					&who,
					liq_token_id.into(),
					non_vested_rewards,
				){
					log!(
						error,
						"Activating liquidity tokens failed upon bootstrap claim rewards = ({:?}, {}, {}), err = {:?}",
						who,
						liq_token_id,
						non_vested_rewards,
						err
					);
					
					Self::deposit_event(Event::RewardsLiquidityAcitvationFailed(who.clone(), liq_token_id, non_vested_rewards));
				};
```

File: pallets/bootstrap/src/lib.rs [hunk 3]
```diff
@@ -157,6 +157,9 @@ pub mod pallet {
 
 		type PoolCreateApi: PoolCreateApi<AccountId = Self::AccountId>;
 
+		#[pallet::constant]
+		type BootstrapUpdateBuffer: Get<Self::BlockNumber>;
```
Comment (mateuszaaa) [inline pallets/bootstrap/src/lib.rs:161]:
```suggestion
		type BootstrapUpdatePeriod: Get<Self::BlockNumber>;
```

i think period would be a better world ?

**Review by mateuszaaa — DISMISSED** [review]
few minor changes

**Comment by majusko** [comment]
3 questions:
what is `cancel_bootstrap` ?
did you remove the `sudo` from finalization of bootstrap? Cannot see it.
did you remove `limit` param in finalization - if not, what is the current default behaviour?

**Review by majusko — COMMENTED** [review]
3 questions:
what is `cancel_bootstrap` ?
did you remove the `sudo` (root) from the finalization of bootstrap? Cannot see it.
did you remove the `limit` param in finalization - if not, what is the current default behaviour?

**Review by majusko — CHANGES_REQUESTED** [review]
3 questions:
what is `cancel_bootstrap` ?
did you remove the `sudo` (root) from the finalization of bootstrap? Cannot see it.
did you remove the `limit` param in finalization - if not, what is the current default behaviour?

**Review by majusko — APPROVED** [review]

File: pallets/bootstrap/src/lib.rs (hunk 3 above)
Comment (iStrike7) [reply pallets/bootstrap/src/lib.rs:161]: Period would indicate that the bootstrap is modifiable within 300 blocks before start, whereas it is the otherway.

File: pallets/bootstrap/src/lib.rs (hunk 2 above)
Comment (iStrike7) [reply pallets/bootstrap/src/lib.rs:888]: Good point. will do

File: pallets/bootstrap/src/lib.rs (hunk 1 above)
Comment (iStrike7) [reply pallets/bootstrap/src/lib.rs:265]: Well, the ask was to "disable the option to..." not "remove the feature..." so I just commented it.

Comment (majusko) [reply pallets/bootstrap/src/lib.rs:265]: Please remove the code as @mateuszaaa suggested. We can always find this in the git history if needed.

**Pull request merged**